src/terraform-resource/terraform/client.go

```diff
@@ -110,6 +115,51 @@ func (c Client) Destroy() error {
 	return nil
 }
 
+
+func (c Client) Plan() error {
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "terraform-resource-client")
+	if err != nil {
+		return fmt.Errorf("Failed to create temporary working dir at '%s'", os.TempDir())
+	}
+	defer os.RemoveAll(tmpDir)
+
+	initCmd := terraformCmd([]string{
+		"init",
+		c.Model.Source,
+		tmpDir,
+	})
+	if initOutput, initErr := initCmd.CombinedOutput(); initErr != nil {
+		return fmt.Errorf("terraform init command failed.\nError: %s\nOutput: %s", initErr, initOutput)
+	}
+
+	getCmd := terraformCmd([]string{
+		"get",
+		"-update",
+		tmpDir,
+	})
+	if getOutput, getErr := getCmd.CombinedOutput(); getErr != nil {
+		return fmt.Errorf("terraform get command failed.\nError: %s\nOutput: %s", getErr, getOutput)
+	}
+
+	planArgs := []string{
+		"plan",
+		"-input=false", // do not prompt for inputs
+		"-out=plan",
```

Comment:
@Seraf Concourse will pass along the contents of the output directory to subsequent tasks in the same job, but I think for this feature we'll need the file available in a separate job that can be manually triggered. To pass files between jobs, we'll have to store it in S3. You can use the [Statefile struct](https://github.com/ljfranklin/terraform-resource/blob/master/src/terraform-resource/terraform/statefile.go) as a reference. It's a wrapper around the StorageDriver class which can upload / download / check for existence from S3.